Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx

```diff
@@ -0,0 +1,680 @@
+      setNoKeys(false);
+      const tokenCredentials = new TokenCredentials(token);
+      const cognitiveServicesManagementClient = new CognitiveServicesManagementClient(tokenCredentials, subscriptionId);
+      const accounts = await cognitiveServicesManagementClient.accounts.list();
+
+      const authoring: KeyRec[] = await fetchKeys(
+        cognitiveServicesManagementClient,
+        accounts.filter((a) => a.kind === 'LUIS.Authoring')
+      );
+      const prediction: KeyRec[] = await fetchKeys(
+        cognitiveServicesManagementClient,
+        accounts.filter((a) => a.kind === 'LUIS')
+      );
+      setLoadingLUIS(false);
+      if (authoring.length == 0 || prediction.length == 0) {
+        setNoKeys(true);
+        setActionOptions([
+          { key: 'create', text: formatMessage('Create a new LUIS resource') },
+          { key: 'handoff', text: formatMessage('Generate a resource request'), disabled: false },
+          { key: 'choose', text: formatMessage('Choose from existing'), disabled: true },
+        ]);
+      } else {
+        setNoKeys(false);
+        setAuthoringKeys(authoring);
+        setPredictionKeys(prediction);
+        setActionOptions([
+          { key: 'create', text: formatMessage('Create a new LUIS resource') },
+          { key: 'handoff', text: formatMessage('Generate a resource request'), disabled: false },
+          { key: 'choose', text: formatMessage('Choose from existing'), disabled: false },
+        ]);
+      }
+    }
+  };
+
+  const fetchResourceGroups = async (subscriptionId) => {
+    if (token) {
+      const tokenCredentials = new TokenCredentials(token);
+      const resourceClient = new ResourceManagementClient(tokenCredentials, subscriptionId);
+      const groups = await resourceClient.resourceGroups.list();
+
+      setResourceGroups([
+        {
+          id: CREATE_NEW_KEY,
+          data: { icon: 'Add' },
+          name: formatMessage('Create new'),
+        },
+        ...groups,
+      ]);
+    }
+  };
+
+  const createLUIS = async () => {
+    let endpointKey = '';
+    let authoringKey = '';
+    if (token) {
+      setLoadingLUIS(true);
+      const tokenCredentials = new TokenCredentials(token);
+
+      const resourceGroupName = resourceGroupKey === CREATE_NEW_KEY ? newResourceGroupName : resourceGroup;
+      if (resourceGroupKey === CREATE_NEW_KEY) {
+        try {
+          const resourceClient = new ResourceManagementClient(tokenCredentials, subscriptionId);
+          await resourceClient.resourceGroups.createOrUpdate(resourceGroupName, {
+            location: localRootLuisRegion,
+          });
+        } catch (err) {
+          setOutcomeDescription(
+            formatMessage(
+              'Due to the following error, we were unable to successfully add your selected LUIS keys to your bot project:'
+            )
+          );
+          setOutcomeSummary(
+            <div>
+              <p>{err.message}</p>
+            </div>
```

Comment:
Not sure we need this `<div>` here either.

Reply:
got it